feature/groups/src/main/java/com/mifos/feature/groupsList/presentation/GroupsListScreen.kt

```diff
@@ -0,0 +1,383 @@
+
+                        is GroupsListState.Error -> {
+                            MifosSweetError(
+                                message = state.message,
+                                onclick = onRefresh,
+                            )
+                        }
+
+                        is GroupsListState.GroupsFromAPI -> {
+                            GroupsListItemFromAPI(
+                                lazyListState = lazyListState,
+                                data = state.groups.collectAsLazyPagingItems(),
+                                doesSelected = {
+                                    selectedItems.contains(it)
+                                },
+                                doesInSelectionMode = {
+                                    selectedItems.isNotEmpty()
+                                },
+                                onGroupClick = onGroupClick,
+                                onSelectItem = {
+                                    if (selectedItems.contains(it)) {
+                                        selectedItems.remove(it)
+                                    }else {
+                                        selectedItems.add(it)
+                                    }
+                                },
+                                onRefresh = onRefresh
+                            )
+                        }
+
+                        is GroupsListState.GroupsFromLocalDB -> {
+                            GroupsListItemFromLocalDB(
+                                lazyListState = lazyListState,
+                                data = state,
+                                doesSelected = {
+                                    selectedItems.contains(it)
+                                },
+                                doesInSelectionMode = {
+                                    selectedItems.isNotEmpty()
+                                },
+                                onGroupClick = onGroupClick,
+                                onSelectItem = {
+                                    if (selectedItems.contains(it)) {
+                                        selectedItems.remove(it)
+                                    }else {
+                                        selectedItems.add(it)
+                                    }
+                                },
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun GroupsListItemFromAPI(
+    modifier: Modifier = Modifier,
+    lazyListState: LazyListState,
+    data: LazyPagingItems<Group>,
+    doesSelected: (Group) -> Boolean,
+    doesInSelectionMode: () -> Boolean,
+    onGroupClick: (Group) -> Unit,
+    onSelectItem: (Group) -> Unit,
+    onRefresh: () -> Unit,
+) {
+    Crossfade(
+        targetState = data.loadState.refresh,
+        label = "Load::States"
+    ) {
+        when (it) {
+            is LoadState.Error -> {
```

Comment:
These and below should merge and should wrapped around the LazyColumn

Reply:
If I wrap LoadingIndicator to inside of LazyColumn then it will visible as single item on top instead Full Screen

Reply:
If i wrapped this around LazyColumn then the LoadingIndicator will visible as single item instead of Full Screen, kindly check it, if it is not suitable then i will wrap this around lazy column.